pkg/channel/test/mock.go

```diff
@@ -2,88 +2,59 @@ package test
 
 import (
 	"errors"
-	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/Shopify/sarama"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
 	kafkav1alpha1 "knative.dev/eventing-contrib/kafka/channel/pkg/apis/messaging/v1alpha1"
 	kafkalisters "knative.dev/eventing-contrib/kafka/channel/pkg/client/listers/messaging/v1alpha1"
-	kafkaproducer "knative.dev/eventing-kafka/pkg/common/kafka/producer"
 )
 
 //
-// Mock Confluent Producer
+// Mock Kafka Producer
 //
 
-var _ kafkaproducer.ProducerInterface = &MockProducer{}
+var _ sarama.SyncProducer = &MockSyncProducer{}
 
-type MockProducer struct {
-	config              *kafka.ConfigMap
-	produce             chan *kafka.Message
-	events              chan kafka.Event
-	testResponseMessage *kafka.Message
-	closed              bool
+type MockSyncProducer struct {
```

Comment:
We have these mocks all around eventing codebase :worried: Maybe not in this pr, but it might be good to start sharing these test utils (better, contribute some of them back to sarama)

Reply:
Yeah, fair enough - I'll try to do more checking in `eventing` and `eventing-contrib` in the future.  Generally though it really slows things down to have to create a separate PR in another repo (if the mocks there aren't adequate) and then push that through before being able to continue with current PR.   How do you guys normally handle such things in Knative?